src/main/java/org/reaktivity/nukleus/http/internal/stream/ServerAcceptStream.java

```diff
@@ -1064,14 +1064,13 @@ private boolean headersMatch(
     {
         boolean[] headersMatch = new boolean[1];
         headersMatch[0] = true;
+
         AtomicReference<String> routeScheme = new AtomicReference<>("");
-        routeHeaders.forEach(routeHeader ->
+        HttpHeaderFW schemeHeader = routeHeaders.matchFirst(header -> ":scheme".equals(header.name().asString()));
+        if (schemeHeader != null)
         {
-            if (routeHeader.name().asString().equals(":scheme"))
-            {
-                routeScheme.set(routeHeader.value().asString());
-            }
-        });
+            routeScheme.set(schemeHeader.value().asString());
+        }
```

Comment:
The intent behind using `matchFirst` to return `schemeHeader` is to remove the need for the `routeScheme` `AtomicReference` - please remove it and use the following instead.
```
final String routeScheme = schemeHeader != null ? schemeHeader.value().asString() : null;
```
then below, use `routeScheme` instead of `routeScheme.get()`.